functions/test_goldens.ts

```diff
@@ -0,0 +1,36 @@
+import * as firebaseFunctions from 'firebase-functions';
+import * as firebaseAdmin from 'firebase-admin';
+import * as path from 'path';
+
+const gcs = require('@google-cloud/storage')();
+
+/** The storage bucket to store the images. The bucket is also used by Firebase Storage. */
+const bucket = gcs.bucket(firebaseFunctions.config().firebase.storageBucket);
+
+/**
+ * Copy files from /screenshot/$prNumber/test/ to goldens/
+ * Only copy the files that test result is failure. Passed test images should be the same as
+ * goldens.
+ */
+export function copyTestImagesToGoldens(prNumber: string) {
+  return firebaseAdmin.database().ref(`screenshot/reports/${prNumber}/results`).once('value')
+    .then((snapshot: firebaseAdmin.database.DataSnapshot) => {
+      let keys: string[] = [];
+      let counter = 0;
+      snapshot.forEach((childSnapshot: firebaseAdmin.database.DataSnapshot) => {
+        if (childSnapshot.val() == false) {
+          keys.push(childSnapshot.key);
+        }
+        counter ++;
+        if (counter == snapshot.numChildren()) return true;
```

Comment:
It is. 
But typescript requires return boolean, and once boolean returned the `forEach` loop will end.